src/main/java/com/shapesecurity/salvation2/Salvation2.java

```diff
@@ -0,0 +1,52 @@
+package com.shapesecurity.salvation2;
+
+import org.teavm.jso.JSBody;
+
+public class Salvation2 {
+	public static void main(String[] args) {
+		initParseList();
+		initParseSingle();
+	}
+
+	public static String parseSerializedCSPList(String policyText) {
```

Comment:
```suggestion
	public static String getErrorsForSerializedCSPList(String policyText) {
```
maybe? A method called "parse" should return the parsed result, not errors.

Ditto other method.

Reply:
Good suggestion!

Reply:
I should point out that this name isn't used in the javascripts. 

Below it is where the function is set on `window`.
````
	@JSBody(params = {}, script =
		"(exports || window).getErrorsForSerializedCSPList = (policyText) => {\n" +
		"return javaMethods.get('com.shapesecurity.salvation2.Salvation2.parseSerializedCSPList(Ljava/lang/String;)Ljava/lang/String;').invoke(policyText)\n" +
		"}")
	static native void initParseList();
```